omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java

```diff
@@ -910,7 +894,65 @@ private void addPaths() {
 		Collections.sort(tagList);
 		swaggerSpecification.setTags(tagList);
 	}
-	
+
+	/**
+	 * Extracts resource name from a resource handler
+	 * @param trim if false returns resource name with version, if true version part of the name is trimmed
+	 * @return resource name
+	 */
+	private String getResourceName(DelegatingResourceHandler resourceHandler, boolean trim) {
+		Resource annotation = resourceHandler.getClass().getAnnotation(Resource.class);
+
+		String resourceName = null;
+		if (annotation != null) {
+			// top level resource
+			resourceName = annotation.name();
+		} else {
+			// sub resource
+			SubResource subResourceAnnotation = resourceHandler.getClass().getAnnotation(SubResource.class);
+			if (subResourceAnnotation != null) {
+				resourceName = subResourceAnnotation.path();
+			}
+		}
+		if (trim) {
+			resourceName = trimResourceName(resourceName);
+		}
+		return resourceName;
+	}
+
+	/**
+	 * Extracts resource parent name from a resource handler
+	 * @param trim if false returns name with version, if true version part of the name is trimmed
+	 * @return name of the parent resource or null if its not a sub-resource
+	 */
+	private String getResourceParentName(DelegatingResourceHandler resourceHandler, boolean trim) {
+		SubResource subResourceAnnotation = resourceHandler.getClass().getAnnotation(SubResource.class);
```

Comment:
So what about my first solution. It's some what efficient as well. getResName() can be used for both Res and SubRes. getResName() uses reflection to check whether it a Resource or a SubResource where as getResourceParentName() pre-assumes that it is a SubResource. If it's not a SubResource then it returns null.

That would be great that if we could get one to delegate other, but getting name and the parent have different conditioning order. So would you want me to keep my current commit or to make a new commit following the single method approach as shown in gist?